src/Dialect/ONNX/ONNXUnsupportedOps.hpp

```diff
@@ -76,3 +64,17 @@ UNSUPPORTED_OPS(ONNXTreeEnsembleClassifierOp)
 UNSUPPORTED_OPS(ONNXTreeEnsembleRegressorOp)
 UNSUPPORTED_OPS(ONNXUpsampleV7Op)
 UNSUPPORTED_OPS(ONNXZipMapOp)
+// These ops were decomposed into supported ops during Decompose pass.
+UNSUPPORTED_OPS(ONNXCastLikeOp)
```

Comment:
Since `CastLikeOp` is unsupported do we still need to have the backend tests I added? Also do we need the version specified at `utils/gen_onnx_mlir.py` for `CastLikeOp`?

Reply:
@hamptonm1 `unsupported` is quite confusing here so I will reorganize this as Alex suggested to make it clearer. 

CastLikeOp is marked `Unsupported` meaning that there is no handling of shape inference/lowering/... inside the compiler because it was rewritten into a "supported" op (i.e. CastOp).

Reply:
Backend tests are still needed. We can run a model that contains CastLikeOp. Just the way we support it is via CastOp instead of directly lowering it to krnl/llvm.

Reply:
Understood it was my assumption that because we are converting CastLikeOp to CastOp that we will no longer need the backend tests. Especially because the backend tests I just added and they are exactly the same for CastOp.